### PR TITLE
[netcore] Fix System.Tests.PseudoCustomAttributeTests.* tests

### PR DESCRIPTION
**Pull request by EgorBo**
~@filipnavara noticed my previous [PR](https://github.com/mono/mono/pull/14056#issuecomment-483387022) broke `System.ComponentModel.Composition.Tests`.
Those were expecting to find custom attributes by an interface those attributes implement.~

This PR also fixes 7 test cases in `System.Tests.PseudoCustomAttributeTests.*`

Now results are:
```
System.Runtime.Tests  Total: 32363, Errors: 0, Failed: 38, Skipped: 0, Time: 28.718s
```
```
System.ComponentModel.Composition.Tests  Total: 1428, Errors: 0, Failed: 2, Skipped: 0, Time: 1.181s
```

`build-local-corefx-test-%` is a command to build corefx tests locally and put them to corefx/tests/extracted/$*.

### CODE REVIEW

**Comment by filipnavara** [comment]
I already a submitted PR to fix the regression (https://github.com/mono/mono/pull/14063). Feel free to close it if this one supersedes it.

File: mcs/class/System.Private.CoreLib/System/Attribute.cs [hunk 1]
```diff
@@ -6,6 +6,14 @@ partial class Attribute
 	{
 		static Attribute GetAttr (ICustomAttributeProvider element, Type attributeType, bool inherit)
 		{
+			if (attributeType == null)
+				throw new ArgumentNullException (nameof (attributeType));
+			
+			if (!attributeType.IsSubclassOf (typeof (Attribute)) &&
+			    attributeType != typeof (Attribute) &&
+			    attributeType != typeof (MonoCustomAttrs))
```
Comment (filipnavara) [inline mcs/class/System.Private.CoreLib/System/Attribute.cs]: When exactly is it `typeof (MonoCustomAttrs)`?

Comment (EgorBo) [reply mcs/class/System.Private.CoreLib/System/Attribute.cs]: https://github.com/mono/mono/blob/b1af4c2726cdc211971efacee7d66a6e447336d2/mcs/class/corlib/System/MonoCustomAttrs.cs#L298 and one more place somewhere, I guess it's used as a marker

**Comment by EgorBo** [comment]
@filipnavara oh, I didn't notice your PR. I guess we can merge yours and the fixes from mine

**Comment by filipnavara** [comment]
Tomorrow I'll start working on getting the CI working so we can avoid this in future :-)

File: mcs/class/System.Private.CoreLib/System.Runtime.InteropServices/MarshalAsAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Runtime.InteropServices
+{
+    [StructLayout(LayoutKind.Sequential)]
+    public sealed partial class MarshalAsAttribute : Attribute
```
Comment (EgorBo) [inline mcs/class/System.Private.CoreLib/System.Runtime.InteropServices/MarshalAsAttribute.cs]: Needs https://github.com/dotnet/coreclr/pull/24014 in mono

**Pull request merged**